lib/evaluation/evaluationClientWrapper.ts

```diff
@@ -378,16 +385,12 @@ export class EvaluationClientWrapper {
     });
   }
 
-  public fillSelectableCredentialsToVerifiableCredentialsMapping(
-    selectResults: SelectResults,
-    verifiableCredentials: VerifiableCredential[]
-  ) {
-    if (selectResults) {
-      selectResults.selectableVerifiableCredentials?.forEach((selectableCredential: VerifiableCredential) => {
-        const foundIndex: number = verifiableCredentials.findIndex(
-          (verifiableCredential) => selectableCredential.id === verifiableCredential.id
-        );
-        selectResults.vcIndexes?.push(foundIndex);
+  public fillSelectableCredentialsToVerifiableCredentialsMapping(selectResults: SelectResults) {
+    const srm: SubmissionRequirementMatch[] = selectResults.matches as SubmissionRequirementMatch[];
+    if (selectResults && srm?.length) {
+      srm[0].matches.forEach((_, index, matches) => {
+        matches[index] = `$[${index}]`;
+        selectResults.vcIndexes?.push(index);
```

Comment:
This is incorrect. I will push the change. 
vcIndex should have an index corresponding to the param passed to selectFromMethod.
This ticket was not supposed to change this function.

Reply:
I noticed unexpected change by reverting that was also not expected. I can push the changes to this branch if you allow. But those will "unexpectedly" brake 5-7 test cases.

Reply:
Actually, I have 3 points here:

1. In the test there are 6 credentials (duplicated array 3x3) from the wallet passed to selectFrom, so Sadjad previous answer [ "$[2]", "$[5]"] is correct;
2. The findIndex function matches the first occurrence, so the actual result of vcIndexes is 2 and 2 instead of 2 and 5 due to vcs duplication;
3. There is no remapping in selectFrom.

Reply:
Fixed!